### ISSUE TITLE
`mapStateToCellProps` does not translate error messages

### ISSUE THREAD

**Issue by codefactor** [eclipsesource/jsonforms#2090]
### Describe the bug

When creating custom Cells, the `errors` prop is always the untranslated message, and not the value returned by the `translateError` function of our `JsonFormsI18nState` object.

### Expected behavior

The `errors` passed by `mapStateToCellProps` should be passed through `translateError` to get project-level translations.

### Steps to reproduce the issue

1. Create your own cell renderer that expects to get the error message
2. Create a form that uses the cell, and define a custom translateError function
3. Check the error message that the cell renderer gets
4. See that the cell does not get the translations from the translateError

### Screenshots

I don't have any screenshots.

### In which browser are you experiencing the issue?

any

### Which Version of JSON Forms are you using?

v3.0.0

### Framework

Core

### RendererSet

Other (please specify in the Additional context field)

### Additional context

We're creating our own custom renderers, where the low-level input controls need the error message. For an example of the input control itself see the following:
https://sap.github.io/ui5-webcomponents-react/?path=/docs/inputs-input--default-story

This component uses a `<ui5-input>` web component which has a `valueStateMessage` field to specify the error message that only displays once you focus into the input and shows as a tooltip on the control.

We've created a custom `TextCell` that uses the above Input control internally; however, we've noticed that none of the messages that show up here are ever translated. Upon investigation this appears to be a bug at the `@jsonforms/core` layer here:
https://github.com/eclipsesource/jsonforms/blob/1b22f55dce2150c29999ad1fb220fc866023a6a0/packages/core/src/util/cell.ts#L151-L153

The only solution/workaround appears to have our `TextCell` to formulate the error message by directly calling `mapStateToControlProps` after pulling the ctx object which is not a clean solution. I suspect that with typical renderers the Cell components don't read the errors prop, so this may never have been brought up before.

**Comment by codefactor** [comment]
https://codesandbox.io/s/romantic-lederberg-63c4td?file=/src/App.jsx
<img width="959" alt="image" src="https://user-images.githubusercontent.com/730310/216840071-98fd4890-a85e-4787-ab06-2d0ab4279728.png">

In the above simple/contrived example, I've created 2 custom renderers - one for a `TextCell` and one for a `TextControl`, the renderer simply outputs the `props.errors` string so that it's visible on the UI. The `<JsonForms>` is passed a simple translate function and translateError function to wrap the default translation with a simple string so that it's obvious whether or not the custom translate function is used to translate a message.

We see in the red box that the text is not being passed through translation, and that we only get the default developer-only english message. But the green box we see the message is wrapped so it's working as expected.

**Comment by codefactor** [comment]
Here is a gist for a function to wrap my custom Cells with to get a translated error message:
https://gist.github.com/codefactor/7fb0c13816b49555f6e9b2a2c23b78fd

To use this solution the Cell 

```typescript
// BEFORE
const CustomTextCell = withJsonFormsCellProps(
  withVanillaCellProps(CustomTextCellRenderer)
);

// AFTER
const CustomTextCell = withJsonFormsCellPropsWithI18nError(
  withVanillaCellProps(CustomTextCellRenderer)
);
```

We can consider the above a kind of hack because it requires the Wrapper logic to do duplicate logic that the mapStateToCellProps has already done.

I believe the true fix here is to call `getCombinedErrorMessage` from `cell.ts`:
https://github.com/eclipsesource/jsonforms/blob/1b22f55dce2150c29999ad1fb220fc866023a6a0/packages/core/src/util/cell.ts#L151-L153

```typescript
  const t = getTranslator()(state);
  const te = getErrorTranslator()(state);
  const errors = getCombinedErrorMessage(errors, te, t, schema, uischema, path);
```

But I guess that the authors here have purposely done this for some reason, and I guess it's because most Cell type renderers don't even read the useless `errors` string - since it won't be translated I'm not sure why it's even there?

**Comment by sdirix** [comment]
This was [discussed in our forum](https://jsonforms.discourse.group/t/why-does-mapstatetocellprops-not-translate-error-messages/1291).

In summary: The `mapStateToCellProps` should invoke the i18n feature too when handing over `errors`.

**Comment by codefactor** [comment]
Probably later today I can propose a PR for this, as the fix is already documented here it should be straight forward.

**Comment by codefactor** [comment]
@sdirix ,
I am not 100% sure how to run the unit tests - but I tried my best to write a test - please see the following PR:
https://github.com/eclipsesource/jsonforms/pull/2098

Any assistance here to get me started running a test?

Here's what I did so far:

1. tried to run `npm install` - get failure about old lock file, so I used nvm to downgrade to version 14 of node (which is out of maintenance now, but I guess you guys still use it... you should probably think to upgrade?)
2. ran `npm install` again on the root project and then `npm run init` and then `npm run test` - all from root project

I am getting errors like this:
```
> @jsonforms/core@3.1.0-alpha.1 test /Users/i832513/SAPDevelop/xweb/jsonforms/packages/core
> cross-env TS_NODE_COMPILER_OPTIONS={\"module\":\"commonjs\",\"target\":\"es5\"} ava


  Uncaught exception in test/generators/uischema.test.ts

  /Users/i832513/SAPDevelop/xweb/jsonforms/node_modules/ts-node/src/util.ts:58

  SyntaxError: Unexpected token a in JSON at position 1
```

I looked through other unit tests for `cell.ts` and I can't find any unit tests that were testing for errors, so I am hoping that the unit tests will pass.

**Comment by sdirix** [comment]
Thanks for the PR!

`npm ci && npm run init && npm run build && npm run test` is usually the way to go. However we have some problems with tests on Mac at the moment which is why they are disabled there in the CI too. The tests will be executed by our CI on Linux and Windows.

> you should probably think to upgrade?)

It's a high priority item for the near future :+1:

**Comment by codefactor** [comment]
@sdirix ,
I only have a Mac to run the unit tests. How do mac users like myself run the tests locally?

**Comment by codefactor** [comment]
@sdirix ,
I tried with a Dockerfile like this:
```
FROM node:14
COPY . .
RUN npm ci && npm run init && npm run build && npm run test
```
Then ran like this:
```
docker build -t jsonforms-test:0.0.1-SNAPSHOT .
```
And the build got further this time but failed:
```
#5 568.5   ✔ util › renderer.ts › mapStateToLabelProps - i18n - default translation has no effect
#5 568.5   ✔ util › renderer.ts › mapStateToLabelProps - i18n - default key translation
#5 568.5   ✔ util › renderer.ts › mapStateToLabelProps - i18n - default message translation
#5 568.5   ✔ util › renderer.ts › mapStateToLabelProps - i18n - i18n key translation
#5 568.5   ✖ test/util/cell.test.ts exited due to SIGABRT
#5 568.5 
#5 568.5   310 tests passed
#5 568.5   1 known failure
#5 568.5 
#5 568.5   generators › schema.ts › default schema generation tuple array types
#5 568.5 
#5 568.5 
#5 568.5 lerna ERR! npm run test stderr:
#5 568.5 
#5 568.5 <--- Last few GCs --->
```
Since it failed on the `cell.test.ts` - I guess it could be a coincidence or maybe it's my code that caused it to fail.

**Comment by codefactor** [comment]
@sdirix ,
Thanks, that's what I suspected -- can you help me to understand the workflow for a mac user to run the tests to debug a problem such as this?

**Comment by sdirix** [comment]
It seems you were able to reproduce the problems in a docker container, right? So in that case you can simply use our [VS Code Dev Container](https://github.com/eclipsesource/jsonforms#vs-code-dev-container)

**Comment by codefactor** [comment]
@sdirix ,

Thanks, I figured out how to make my container approach re-run the tests without npm install every time, which was my issue before. I will consider giving the VS Code Dev Container a try if I do more contributions to this project.

**Comment by sdirix** [comment]
Fixed with #2098
